types/lodash/ts3.1/common/util.d.ts

```diff
@@ -752,6 +754,10 @@ declare module "../index" {
          * @param predicates The predicates to check.
          * @return Returns the new function.
          */
+        overEvery<A, B extends A, C extends B>(
+            fn1: Refinement<A, B>,
+            fn2: Refinement<B, C>,
+        ): Refinement<A, C>;
         overEvery<T>(...predicates: Array<Many<(...args: T[]) => boolean>>): (...args: T[]) => boolean;
```

Comment:
If we're happy with this, I'll go ahead and add the same overload for the other variants.

I can also extend support for the number of allowed functions, if need be.

Reply:
Looks good to me, go ahead and d othe other variants.
Let's stick with just 2 arguments for now, unless a need arises for more arguments.

Reply:
I just tried to add the other variants, but realised it's not possible.

IIUC, the chain variants must accept only one parameter. I could add an overload for a tuple?

When I ran the FP generate script, no changes were made 🤷‍♂

Reply:
That's fair. The FP generate probably didn't do anything because it's expecting a 1-argument overload. Adding an overload for tuple would probably fix FP. Maybe to keep things simple we should *only* support the tuple version? Just to prevent things from getting out of hand if we ever add overloads for 3+ arguments.

Is it even possible to do tuples for the chained version? I think in the 3.1 version, you'll just end up with `CollectionChain<T>` which you can't really make into a tuple. If so, feel free to ignore the chains.

Reply:
> Maybe to keep things simple we should _only_ support the tuple version?

I agree. However, as soon as I switched to the tuple version, my test failed with:

```
ERROR: 6851:5  expect  TypeScript@next expected type to be:
  Refinement<{}, Named & Aged>
got:
  Refinement<{}, Aged>
```

😕 

I have no idea why that would be happening. In VS Code, with the TS selected to use the workspace version, the return type looks fine

![image](https://user-images.githubusercontent.com/921609/61355882-26f4f480-a86d-11e9-84c6-c943dc71e981.png)

Reply:
> Adding an overload for tuple would probably fix FP. 

For some reason, `npm run generate` still produced no changes after I had switched to use a tuple instead.

Reply:
Hi @aj-r, any ideas on the above? I will re-open a new PR once we've decided how to proceed.

Reply:
Sorry, I'm not sure what's up with that - I don't see why tuples would behave differently (but I'm also not an expert on the TS compiler).

You could try returning `Refinement<A, B & C>` - I feel like that should fix it (and is probably more reliable anyway).

```ts
overEvery<A, B extends A, C extends A>(fns: [
    Refinement<A, B>,
    Refinement<B, C>,
]): Refinement<A, B & C>;
```

I'm not sure what's up with `generate-fp`. You may need to debug it :/. I find that conditional breakpoints help a lot. If you're using VS Code, I've created a `launch.json` file that should make debugging work, as long as VS Code's project root is `DefinitelyTyped/types/lodash`.

Also keep in mind that there are now 2 `generate` scripts: one for TS<3.1 and one for TS>=3.1.